test/e2e/test/logstash/checks.go

```diff
@@ -143,6 +143,20 @@ func CheckStatus(b Builder, k *test.K8sClient) test.Step {
 	}
 }
 
+func uniqueAssociationCount(refs, otherRefs []v1.ObjectSelector) int {
+	uniqueAssociations := make(map[v1.ObjectSelector]bool)
+
+	for _, val := range refs {
+		uniqueAssociations[val] = true
+	}
+
+	for _, val := range otherRefs {
+		uniqueAssociations[val] = true
+	}
+
+	return len(uniqueAssociations)
+}
```

Comment:
nit: using a `struct{}` instead of a `bool` is more optimal because the former doesn't use any additional memory (I do however agree that it doesn't make much difference in this case since we are dealing with a very small set 😄 )

```suggestion
func uniqueAssociationCount(refsList ...[]v1.ObjectSelector) int {
	uniqueAssociations := make(map[v1.ObjectSelector]struct{})
	for _, refs := range refsList {
		for _, val := range refs {
			uniqueAssociations[val] = struct{}{}
		}
	}
	return len(uniqueAssociations)
}
```